scripts/ci/install

```diff
@@ -13,6 +14,13 @@ os.chdir(REPO_ROOT)
 def run(command):
     return check_call(command, shell=True)
 
+if sys.version_info[:2] >= (3, 12):
+    # Python 3.12+ no longer includes setuptools by default.
+
+    # Setuptools 71+ now prefers already installed versions
+    # of packaging _and_ broke the API for packaging<22.0.
+    # We'll pin to match what's in requirements-dev.txt.
+    run("pip install setuptools==71.1.0 packaging==24.1")
```

Comment:
Why do we need to pin `packaging`? Wouldn't a compatible version be installed when we pip install `setuptools==71.1.0`?

Reply:
Ported change from python 3.12 support for v1.